### PR TITLE
fix routeEntryComplete state leakage (#543).

### PR DESCRIPTION
**Pull request by kbullaughey**

### CODE REVIEW

**Comment by marcoow** [comment]
I think you can remove the test as it doesn't really test anything anyways.

**Comment by kbullaughey** [comment]
Removed the test.

File: packages/ember-simple-auth/lib/simple-auth/mixins/application-route-mixin.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ import Configuration from './../configuration';
 
 var routeEntryComplete = false;
 
+var resetApplicationRouteMixinInternalState = function() {
+  routeEntryComplete = false;
+  return 'reset';
```
Comment (marcoow) [inline packages/ember-simple-auth/lib/simple-auth/mixins/application-route-mixin.js]: this can be removed

**Comment by marcoow** [comment]
I think it's better to store this in a property of the application route that in the local variable anyway. That way we can also have a simple method in `setup.js` that unsets it and don't need the extra export in `application-route-mixin.js`. Sorry for the change in mind.

**Comment by kbullaughey** [comment]
I think as a property on the application route there's no need to reset it during setup because the route object should be created fresh for each acceptance and unit test, no? 

I've pushed a new version to my branch.

File: packages/ember-simple-auth/lib/simple-auth/mixins/application-route-mixin.js [hunk 2]
```diff
@@ -37,13 +35,18 @@ var routeEntryComplete = false;
   @extends Ember.Mixin
   @static
 */
-export default Ember.Mixin.create({
+var ApplicationRouteMixin = Ember.Mixin.create({
```
Comment (marcoow) [inline packages/ember-simple-auth/lib/simple-auth/mixins/application-route-mixin.js]: Can you change this back to `export default` so it's consistent with the other files again as the extra var isn't needed anymore?

**Comment by marcoow** [comment]
You're right of course and that makes it even simpler. Can you add a test?

**Comment by kbullaughey** [comment]
Okay. Updated the tests and fixed the export statement.

File: packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js [hunk 3]
```diff
@@ -44,6 +44,7 @@ describe('ApplicationRouteMixin', function() {
       });
 
       it("translates the session's 'sessionAuthenticationSucceeded' event into an action invocation", function(done) {
+        expect(this.route.get('_authRouteEntryComplete')).to.be.true;
```
Comment (marcoow) [inline packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js]: Sorry for being so finicky, I actually really appreciate the PR, but this should have it's own test…

File: packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js [hunk 4]
```diff
@@ -52,6 +53,23 @@ describe('ApplicationRouteMixin', function() {
...
+        route2 = Ember.Route.extend(ApplicationRouteMixin, {
+          transitionTo: function() {}
+        }).create({ session: this.session });
+        var transition = { send: function() {} };
+        sinon.spy(transition, 'send');
+        sinon.spy(route2, 'send');
+        route2.beforeModel(transition);
+        this.session.trigger('sessionAuthenticationSucceeded');
+        Ember.run.next(this, function() {
+          expect(route2.get('_authRouteEntryComplete')).to.not.be.true;
```
Comment (marcoow) [inline packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js]: no need to assert on this here.

Comment (kbullaughey) [inline packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js]: Fixed.

File: packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js (hunk 3 above)
Comment (kbullaughey) [inline packages/ember-simple-auth/tests/simple-auth/mixins/application-route-mixin-test.js]: No worries. Fixed.

**Comment by marcoow** [comment]
:+1:

**Pull request merged**